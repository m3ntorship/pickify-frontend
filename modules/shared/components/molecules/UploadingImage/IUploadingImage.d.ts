declare namespace IUploadingImage {
  export interface IProps {
    imageUrl: string;
  }
}
export { IUploadingImage };
