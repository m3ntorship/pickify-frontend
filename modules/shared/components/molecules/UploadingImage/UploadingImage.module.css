.container {
  @apply flex flex-col;
}
