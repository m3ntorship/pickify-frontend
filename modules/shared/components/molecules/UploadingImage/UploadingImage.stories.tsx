import React from 'react';
import type { Story, Meta } from '@storybook/react';
import UploadingImage from './UploadingImage';
import type { IUploadingImage } from './IUploadingImage';

export default {
  title: 'molecules/UploadingImage',
  component: UploadingImage,
  argTypes: {},
} as Meta;

const Template: Story<IUploadingImage.IProps> = (args) => (
  <UploadingImage {...args} />
);

export const Default = Template.bind({});
Default.args = {
  textInputLetter: 'a',
  id: '1',
};
