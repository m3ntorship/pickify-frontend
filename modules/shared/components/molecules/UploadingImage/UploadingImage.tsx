import React from 'react';
import type { FC, ReactElement } from 'react';
import Image from 'next/image';
import type { IUploadingImage } from './IUploadingImage';
import styles from './UploadingImage.module.css';
import TextInput from '../../atoms/TextInputs/TextInput';
import * as ETextInput from '../../atoms/TextInputs/types/ETextInput';
import VerticalThreeDots from '../../icons/verticalThreeDots.svg';

const UploadingImage: FC<IUploadingImage.IProps> = ({
  imageUrl = 'https://source.unsplash.com/random',
}): ReactElement => {
  return (
    <div className={styles.container}>
      <div className="relative">
        <Image
          src={imageUrl}
          layout="fill"
          objectFit="contain"
          className="w-full h-full rounded-t-md mb-1 "
        />
        <div className="absolute top-4 right-4">
          <VerticalThreeDots />
        </div>
      </div>
      <TextInput
        variants={ETextInput.Variants.Default}
        inputType={ETextInput.InputType.Choices}
        letter="A"
        id="1"
      />
    </div>
  );
};

export default UploadingImage;
