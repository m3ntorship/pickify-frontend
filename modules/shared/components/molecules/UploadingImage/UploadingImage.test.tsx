import * as React from 'react';
import * as renderer from 'react-test-renderer';
import UploadingImage from './UploadingImage';

describe('Testing component with snapshot', () => {
  it('Testing component with required props', () => {
    const tree = renderer.create(<UploadingImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
