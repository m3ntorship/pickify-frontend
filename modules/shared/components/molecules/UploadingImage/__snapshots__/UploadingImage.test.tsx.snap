// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component with snapshot Testing component with required props 1`] = `
<div
  className="container"
>
  <div
    className="relative"
  >
    <div
      style={
        Object {
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "left": 0,
          "margin": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <img
        className="w-full h-full rounded-t-md mb-1 "
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="absolute top-4 right-4"
    >
      <svg
        data-file-name="SvgVerticalThreeDots"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
     
    <div
      className="form-control"
    >
      <input
        className="form-input input-with-choices"
        data-input-type="choices"
        data-testid="text-input"
        data-variant="default"
        id="1"
        onChange={[Function]}
        type="text"
        value=""
      />
      <span
        className="status-icon"
      >
        
      </span>
      <span
        className="control-icon"
      >
        <span
          className="letter"
        >
          A
        </span>
      </span>
    </div>
  </div>
</div>
`;
